cmd/soroban-cli/src/config/self_outdated_check.rs

```diff
@@ -0,0 +1,82 @@
+use crate::config::locator;
+use jsonrpsee_core::Serialize;
+use semver::Version;
+use serde::Deserialize;
+use std::fs;
+
+const FILE_NAME: &str = "self_outdated_check.toml";
```

Comment:
This is really minor, but I think we should call this check the 'update_check'. The phrase rolls off the tongue a little easier.
```suggestion
const FILE_NAME: &str = "upgrade_check.toml";
```